BackofficeBundle/Form/Type/Component/ContentChoiceType.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace OpenOrchestra\BackofficeBundle\Form\Type\Component;
+
+use OpenOrchestra\ModelInterface\Repository\ContentRepositoryInterface;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Form\FormBuilderInterface;
+use OpenOrchestra\BaseBundle\Context\CurrentSiteIdInterface;
+use OpenOrchestra\BackofficeBundle\Form\DataTransformer\ReferenceToEmbedTransformer;
+
+/**
+ * Class ContentChoiceType
+ */
+class ContentChoiceType extends AbstractType
+{
+    protected $contentRepository;
+    protected $contextManager;
+    protected $formTypeName;
+
+    /**
+     * @param ContentRepositoryInterface  $contentRepository
+     * @param CurrentSiteIdInterface      $contextManager
+     * @param ReferenceToEmbedTransformer $referenceToEmbedTransformer
+     * @param string                      $formTypeName
+     */
+    public function __construct(ContentRepositoryInterface $contentRepository, CurrentSiteIdInterface $contextManager, ReferenceToEmbedTransformer $referenceToEmbedTransformer, $formTypeName)
+    {
+        $this->contentRepository = $contentRepository;
+        $this->contextManager = $contextManager;
+        $this->referenceToEmbedTransformer = $referenceToEmbedTransformer;
+        $this->formTypeName = $formTypeName;
+
+        $this->referenceToEmbedTransformer->setFormTypeName($this->formTypeName);
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array                $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addModelTransformer($this->referenceToEmbedTransformer);
+        $builder->add($this->formTypeName, 'choice', array(
+            'label' => false,
+            'choices' => $this->getChoices($options['content_type'], $options['operator'], $options['keyword']),
```

Comment:
I think that you should add a check if the key is present in the array.

If it's not, you should return an exception